src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -409,15 +409,23 @@ ValidatorList::for_each_listed (
 }
 
 std::size_t
-ValidatorList::calculateQuorum (std::size_t nTrustedKeys)
+ValidatorList::calculateMinimumQuorum (
+    std::size_t nListedKeys, bool unlistedLocal)
 {
-    // Use 80% for large values of n, but have special cases for small numbers.
-    constexpr std::array<std::size_t, 10> quorum{{ 0, 1, 2, 2, 3, 3, 4, 5, 6, 7 }};
+    // Use 2f + 1 for large values of n, but have special cases for small numbers.
+    constexpr std::array<std::size_t, 6> quorum{{ 0, 1, 2, 2, 3, 3 }};
```

Comment:
The comment is confusing... first, there's no `n`, and I see nothing about `2f+1` anywhere in the code, although I do see what looks like `2f/3 + 1`. Also, it says we have special cases, but doesn't explain why. But I digress...

If we are _REALLY_ using `2f+1` for _LARGE_ values of `n` then there's no reason for special casing small numbers, since `2f+1` gives the following:

0: 1
1: 1
2: 2
3: 2
4: 3
5: 3

With the exception of `quorum[0]` this matches perfectly with our "special cases". And I am curious as to why we return `0`. That only seems possible in the case where `nListedKeys == 0 && unlistedLocal == false`, so what case is that? Standalone mode? Startup of the first validator of a fresh network?